apps/bookmark/js/bookmark_editor.js

```diff
@@ -19,20 +21,31 @@ var BookmarkEditor = {
   },
 
   _init: function bookmarkEditor_init(mode) {
+    var _ = navigator.mozL10n.get;
     this.mode = document.body.dataset.mode = mode;
     this.bookmarkTitle = document.getElementById('bookmark-title');
     this.bookmarkIcon = document.getElementById('bookmark-icon');
     this.cancelButton = document.getElementById('cancel-button');
     this.saveButton = document.getElementById('done-button');
+    this.appInstallationSection = document.getElementById('app-installation');
+    this.appNameText = document.getElementById('app-name');
+    this.installAppButton = document.getElementById('install-app-button');
 
     this.cancelButton.addEventListener('click', this.close.bind(this));
     this.saveListener = this.save.bind(this);
     this.saveButton.addEventListener('click', this.saveListener);
+    this.installAppButton.addEventListener('click',
+      this._installApp.bind(this));
 
     this.bookmarkTitle.value = this.data.name || '';
 
     this._renderIcon();
 
+    if (this.data.manifestURL) {
+      this.manifestURL = this.data.manifestURL;
+      this._getManifest(this.manifestURL);
```

Comment:
Hmm, I was using "get" in the sense of HTTP GET, but I can see how that could be confusing. I changed it to "_fetchManifest"